libraries/AP_RangeFinder/AP_RangeFinder_Ainstein_LR_D1.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include "AP_RangeFinder_config.h"
+
+#if AP_RANGEFINDER_AINSTEIN_LR_D1_ENABLED
+
+/*
+ questions:
+ - bug on page 22 on malfunction codes
+ - fixed length seems strange at 28 bytes
+ - definition of snr field is missing in documentation
+ - roll/pitch limits are in conflict, 3.2 vs 
+*/
+
+#include "AP_RangeFinder.h"
+#include "AP_RangeFinder_Backend_Serial.h"
+
+class AP_RangeFinder_Ainstein_LR_D1 : public AP_RangeFinder_Backend_Serial
+{
+
+public:
+
+    static AP_RangeFinder_Backend_Serial *create(
+        RangeFinder::RangeFinder_State &_state,
+        AP_RangeFinder_Params &_params) {
+        return new AP_RangeFinder_Ainstein_LR_D1(_state, _params);
+    }
+
+protected:
+
+    MAV_DISTANCE_SENSOR _get_mav_distance_sensor_type() const override {
+        return MAV_DISTANCE_SENSOR_RADAR;
+    }
+
+    uint32_t initial_baudrate(uint8_t serial_instance) const override {
+        return 115200;
+    }
+
+private:
+
+    using AP_RangeFinder_Backend_Serial::AP_RangeFinder_Backend_Serial;
+
+    // get a reading
+    bool get_reading(float &reading_m) override;
+
+    // 0 is no return value, 100 is perfect.  false means signal
+    // quality is not available
+    int8_t get_signal_quality_pct() const override;
+
+    // find signature byte in buffer starting at start, moving that
+    // byte and following bytes to start of buffer.
+    bool move_signature_in_buffer(uint8_t start);
+
+    void check_for_malfunction();
+
+    enum class MalfunctionAlert {
+        Temperature = 0x01,
+        Voltage = 0x02,
+        IFSignalSaturation = 0x40,
+        AltitudeReading = 0x80,
+    };
```

Comment:
If an enum is actually s bitmask, better to make to more obvious by using (1U<<0), (1U<<1), (1U<<2)... instead of 0x1 0x2, 0x4.. and then have them all lined up i with proper whitespace makes it very obvious at a glance that it's a bitwise mask.

Also, adding ` : uint8_t` to the end of the name might make casting easier